### PR TITLE
WT-5040 Add dumping of cursor's page on a mismatch.

### PR DESCRIPTION
**Pull request by sueloverso**
@keithbostic for any suggestions of other/additional places to call this new function. Not yet ready for merge.

### CODE REVIEW

File: test/format/format.h [hunk 1]
```diff
@@ -73,6 +73,7 @@ typedef struct {
 	char *home_rand;			/* RNG log file path */
 	char *home_salvage_copy;		/* Salvage copy command */
 	char *home_stats;			/* Statistics file path */
+	char *home_treedump;			/* Treedump filename */
```
Comment (keithbostic) [inline test/format/format.h]: Possible name change to reflect we're not dumping a tree, we're dumping a page?

File: src/btree/bt_debug.c [hunk 2]
```diff
@@ -657,6 +657,30 @@ __wt_debug_tree_all(
...
+	cbt = (WT_CURSOR_BTREE *)cursor;
+
+	/*
+	 * Dump only the cursor referenced page.
+	 */
+	WT_WITH_BTREE(session, cbt->btree, ret = __wt_debug_page(
+	    session, NULL, cbt->ref, ofile));
+	return (ret);
+}
+
```
Comment (keithbostic) [inline src/btree/bt_debug.c]: Isn't this the same as `wt_debug_cursor_page()`?

File: test/format/snap.c [hunk 3]
```diff
@@ -178,6 +178,11 @@ snap_verify(WT_CURSOR *cursor, TINFO *tinfo, SNAP_OPS *snap)
 	}
 
 	/* Things went pear-shaped. */
+#ifdef HAVE_DIAGNOSTIC
+	fprintf(stderr,
+	    "snapshot-isolation error: Dumping tree to %s\n", g.home_treedump);
+	(void)__wt_debug_tree_cursor(cursor, g.home_treedump);
+#endif
```
Comment (keithbostic) [inline test/format/snap.c]: This is going to hit some of the "function return value unchecked" complaints, should have `testutil_check()`.

File: test/format/snap.c [hunk 4]
```diff
@@ -181,7 +182,9 @@ snap_verify(WT_CURSOR *cursor, TINFO *tinfo, SNAP_OPS *snap)
 #ifdef HAVE_DIAGNOSTIC
 	fprintf(stderr,
 	    "snapshot-isolation error: Dumping tree to %s\n", g.home_treedump);
-	(void)__wt_debug_tree_cursor(cursor, g.home_treedump);
+	t_ret = __wt_debug_tree_cursor(cursor, g.home_treedump);
+	if (ret == 0)
+		ret = t_ret;
```
Comment (keithbostic) [inline test/format/snap.c]: I'd go with `testutil_check()`, else `WT_TRET`.

File: src/btree/bt_debug.c (hunk 2 above)
Comment (sueloverso) [reply src/btree/bt_debug.c]: Yes! After I made the change, I didn't realize the same thing was just a bit further down. I'll fix.

File: test/format/format.h (hunk 1 above)
Comment (sueloverso) [reply test/format/format.h]: I like that.

File: test/format/snap.c (hunk 4 above)
Comment (sueloverso) [reply test/format/snap.c]: Fixed.

File: test/format/util.c [hunk 5]
```diff
@@ -348,6 +348,12 @@ path_setup(const char *home)
 	testutil_check(__wt_snprintf(
 	    g.home_stats, len, "%s/%s", g.home, "stats"));
 
+	/* Tree dump file. */
+	len = strlen(g.home) + strlen("pagedump") + 2;
+	g.home_pagedump = dmalloc(len);
+	testutil_check(__wt_snprintf(
+	    g.home_pagedump, len, "%s/%s", g.home, "pagedump"));
+
```
Comment (keithbostic) [inline test/format/util.c]: Not sure the comment makes sense any more.

Comment (sueloverso) [reply test/format/util.c]: Fixed

**Pull request merged**